### PR TITLE
Added bet settlement functionality

### PR DESCRIPTION
**Pull request by itishg**

### CODE REVIEW

File: backend/src/routers/bets.py [hunk 1]
```diff
@@ -218,4 +218,31 @@ async def get_odds(uid:str,
     documents = await cursor.to_list(length=limit)
     odds_collection = [OddsScheme(odds=float(doc["odds"]), timestamp=int(doc["timestamp"])) for doc in documents]
     return OddsResponse(success=Success(ok=True, error=None, message=""),
-                        odds=odds_collection)
\ No newline at end of file
+                        odds=odds_collection)
+
+@router.post("/settle")
```
Comment (ianmkim) [inline backend/src/routers/bets.py:223]: who is calling this? when is it being called?

File: backend/src/routers/bets.py [hunk 2]
```diff
@@ -218,4 +218,31 @@ async def get_odds(uid:str,
...
+async def settle_bet(settlement: BetSettlement, db=Depends(get_db), mongo=Depends(get_mongo)) -> Success:
+    # Retrieve all wagers for the given bet_uuid
+    wagers_cursor = mongo[DB_BETS].find({"bet_uuid": settlement.bet_uuid})
+    wagers = await wagers_cursor.to_list()
+
+    # Iterate through each wager to calculate and update user balance
```
Comment (ianmkim) [inline backend/src/routers/bets.py:229]: how did you test this? did it work?

File: backend/src/routers/bets.py [hunk 3]
```diff
@@ -218,4 +218,31 @@ async def get_odds(uid:str,
...
+        elif (not wager['yes'] and not settlement.outcome):
+            payout = wager['tokens'] / (1- odds)
+        else:
+            payout = 0
+
+        # Update the user balance
+        user.balance += payout
+        db.commit()
+
+    return Success(ok=True, message="All bets settled and balances updated.")
```
Comment (ianmkim) [inline backend/src/routers/bets.py:248]: This will fail with a validation error, the error=None should be here

File: backend/src/routers/bets.py [hunk 4]
```diff
@@ -218,4 +218,31 @@ async def get_odds(uid:str,
...
+    # Iterate through each wager to calculate and update user balance
+    for wager in wagers:
+        user = get_user(wager['user_uuid'], db)
+        if not user:
+            return Success(ok=False, error=str(), message="Failed to create bet") 
+
+        # Calculate payout based on the odds and whether the user bet Yes or No
+        odds = wager['odds']
+        if (wager['yes'] and settlement.outcome):
+            payout = wager['tokens'] / odds
```
Comment (ianmkim) [inline backend/src/routers/bets.py:238]: what about the original bet amount?

File: backend/src/schemas/bets.py [hunk 5]
```diff
@@ -41,4 +41,7 @@ class WagerCreateContext(BaseModel):
     amount: int
     yes: bool
     bet_uuid: str
-    
\ No newline at end of file
+
+class BetSettlement(BaseModel):
+    bet_uuid: str
+    outcome: bool
```
Comment (ianmkim) [inline backend/src/schemas/bets.py:47]: where is this outcome determined

**Review by ianmkim — CHANGES_REQUESTED** [review]

File: backend/src/routers/bets.py (hunk 4 above)
Comment (itishg) [reply backend/src/routers/bets.py:238]: doesn't wager['tokens'] equal the number of tokens bet by user for that wager?

File: backend/src/schemas/bets.py (hunk 5 above)
Comment (itishg) [reply backend/src/schemas/bets.py:47]: You would be passing in a BetSettlment object when calling /settle which should have the uuid & outcome

Comment (itishg) [reply backend/src/schemas/bets.py:47]: The outcome is determined manually for MVP?

File: backend/src/routers/bets.py (hunk 3 above)
Comment (itishg) [reply backend/src/routers/bets.py:248]: done

**Review by ianmkim — APPROVED** [review]

**Pull request merged**